enhancements/network/ovn-k-localnet-api.md

```diff
@@ -0,0 +1,507 @@
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: test-net
+  namespace: red
+finalizers:
+ - k8s.ovn.org/user-defined-network-protection
+labels:
+  k8s.ovn.org/user-defined-network: ""
+ownerReferences:
+- apiVersion: k8s.ovn.org/v1
+  blockOwnerDeletion: true
+  controller: true
+  kind: ClusterUserDefinedNetwork
+  name: test-net
+spec:
+    config: > '{
+        "cniVersion": "1.0.0",
+        "type": "ovn-k8s-cni-overlay"
+        "netAttachDefName": "blue/test-net",
+        "role": "secondary",
+        "topology": "localnet",
+        "name": "cluster.udn.test-net",
+        "physicalNetworkName: "tenantblue",
+        "subnets": "192.168.0.0/16,2001:dbb::/64",
+        "allowPersistentIPs": true,
+        "excludeSubents: "10.100.50.0/24",
+        "mtu": 9000,
+        "vlanID": 200
+    }'
+```
+
+#### Go types
+The CUDN `spec.network.topology` field should be extended to accept `Localnet` string.
+And the `spec.network` struct `NetworkSpec` should be to have additional field for the localnet topology  configuration:
+```go
+const NetworkTopologyLocalnet NetworkTopology = "Localnet"
+
+...
+
+// NetworkSpec defines the desired state of UserDefinedNetworkSpec.
+// +union
+type NetworkSpec struct {
+    // Topology describes network configuration.
+    //
+    // Allowed values are "Layer3", "Layer2" and "Localnet".
+    // Layer3 topology creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+    // Layer2 topology creates one logical switch shared by all nodes.
+    // Localnet topology attach to the overlay local network. Enables egress to the provider's physical network. 
+    //
+    // +kubebuilder:validation:Required
+    // +required
+    // +unionDiscriminator
+    Topology NetworkTopology `json:"topology"`
+    
+    ...
+    
+    // Localnet is the Localnet topology configuration.
+    // +optional
+    Localnet *LocalnetConfig `json:"localnet,omitempty"`
+}
+```
+
+The CUDN spec should have additional validation rule for `spec.network.topology` field:
+```go
+// ClusterUserDefinedNetworkSpec defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkSpec struct {
+    ...
+    // +required
+    Network NetworkSpec `json:"network"`
+}
+```
+
+As of today the CUDN has validation to ensure spec.network is immutable:
+```go
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Network spec is immutable"
+```
+This validation should be changed to allow mutation localnet topology configurations, at least MTU and VLAN.
+
+Localnet topology configuration type 
+```go
+type LocalnetConfig struct {
+    // Role describes the network role in the pod.
+    //
+    // Allowed value is "Secondary".
+    // Secondary network is only assigned to pods that use `k8s.v1.cni.cncf.io/networks` annotation to select given network.
```

Comment:
What happens when this value is set to `Secondary`? 

How does this annotation information relate to other fields in this API? If `k8s.v1.cni.cncf.io/networks` is set to the value of the `physicalNetworkName` field is this network assigned to the pod?